src/wallet/coinselection.cpp

```diff
@@ -354,6 +355,10 @@ void OutputGroup::Insert(const COutput& output, size_t ancestors, size_t descend
     // descendants is the count as seen from the top ancestor, not the descendants as seen from the
     // coin itself; thus, this value is counted as the max, not the sum
     m_descendants = std::max(m_descendants, descendants);
+
+    if (output.input_bytes > 0) {
+        m_weight += output.input_bytes * WITNESS_SCALE_FACTOR;
```

Comment:
While this is fine for the purposes of this PR, the weight here is not always precise due to rounding error of `input_bytes`.

Ideally, we would store the actual weight in `COutput` and just take it from there. I realise, this might be out of scope for this PR. But maybe we can add a warning in code for future devs and create an issue in this repo?

Reply:
> the weight here is not always precise due to rounding error of input_bytes

Could elaborate, since both `COutput::input_bytes` and `WITNESS_SCALE_FACTOR` are integer ?

Reply:
1 virtual byte = 4 weight units. When computing vbytes from weight, if the weight is not divisible by 4, the vbytes is rounded up. For example, 6 WU is really 1.5 vbytes, but is rounded to 2 vbytes. Since `input_bytes` stores this rounded number, computing weight from it results in 8 WU rather than the correct value of 6.